src/t_string.c

```diff
@@ -160,6 +161,19 @@ void setCommand(client *c) {
 
     c->argv[2] = tryObjectEncoding(c->argv[2]);
     setGenericCommand(c,flags,c->argv[1],c->argv[2],expire,unit,NULL,NULL);
+
+    /* Propagate without the GET argument */
+    if (flags & OBJ_SET_GET) {
+        int argc = 0;
+        robj **argv = zmalloc((c->argc-1)*sizeof(robj*));
```

Comment:
A minor comment, redis flags usually can be duplicated. So SET A B GET GET (More likely a typo like SET A B GET EX 5 GET) is valid, but will get replicated as a SET A B GET.

Reply:
good catch. i'm guessing we may be ok to let it slide (the concern about not replicating the GET is not that important).
But let me try to fix it and see how much uglier it makes the code.
one option is to allocate the argv and accumulate to it right at the top of the function (don't like it), and the other option is to do another string matching check inside the loop here. i think i'll take the second one.